src/lib/viewers/BaseViewer.js

```diff
@@ -1087,7 +1089,13 @@ class BaseViewer extends EventEmitter {
      * @return {void}
      */
     handleAnnotationControlsEscape() {
-        this.annotator.toggleAnnotationMode(AnnotationMode.NONE);
+        if (this.options.enableAnnotationsDiscoverability) {
+            this.annotator.toggleAnnotationMode(AnnotationMode.REGION);
+            this.annotationControlsFSM.reset();
+            this.processAnnotationModeChange(AnnotationMode.NONE);
```

Comment:
The idea behind this change is that we not only want to call `toggleAnnotationMode`, we also want to reset `annotationControlsFSM` currentState because that is the source of truth. Without the above changes, we would end up seeing buggy behavior.

Reply:
Also, since L1093 -> L1095 is the same as L597 -> L599, I could move those out to a shared method - would be open to naming ideas for that method.